rosidl_generator_c/cmake/rosidl_generator_c_generate_interfaces.cmake

```diff
@@ -116,81 +118,81 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     C_STANDARD 11
     COMPILE_OPTIONS -Wall -Wextra -Wpedantic)
 endif()
-if(WIN32)
-  target_compile_definitions(${rosidl_generate_interfaces_TARGET}${_target_suffix}
-    PRIVATE "ROSIDL_GENERATOR_C_BUILDING_DLL_${PROJECT_NAME}")
-endif()
+set_property(TARGET ${rosidl_generate_interfaces_TARGET}${_target_suffix}
+  PROPERTY DEFINE_SYMBOL "ROSIDL_GENERATOR_C_BUILDING_DLL_${PROJECT_NAME}")
```

Comment:
@sloretz unconditionally? I'm OK with that, though talking about DLLs outside Windows is awkward.

Reply:
Looks that way. If you build `rcl` with `-DCMAKE_VERBOSE_MAKEFILE=ON` all the compiler invocations have `-Drcl_EXPORTS` passed to them; which is the default value of `DEFINE_SYMBOL`.